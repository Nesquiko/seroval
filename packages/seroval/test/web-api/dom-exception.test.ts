import { describe, it, expect } from 'vitest';
import {
  crossSerialize,
  crossSerializeAsync,
  crossSerializeStream,
  deserialize,
  fromJSON,
  serialize,
  serializeAsync,
  toJSON,
  toJSONAsync,
} from '../../src';

const EXAMPLE_MESSAGE = 'This is an example message.';
const EXAMPLE_NAME = 'Example';

describe('DOMException', () => {
  describe('serialize', () => {
    it('supports DOMException', () => {
      const example = new DOMException(EXAMPLE_MESSAGE, EXAMPLE_NAME);
      const result = serialize(example);
      expect(result).toMatchSnapshot();
      const back = deserialize<DOMException>(result);
      expect(back).toBeInstanceOf(DOMException);
      expect(back.message).toBe(example.message);
      expect(back.name).toBe(example.name);
    });
  });
  describe('serializeAsync', () => {
    it('supports DOMException', async () => {
      const example = new DOMException(EXAMPLE_MESSAGE, EXAMPLE_NAME);
      const result = await serializeAsync(Promise.resolve(example));
      expect(result).toMatchSnapshot();
      const back = await deserialize<Promise<DOMException>>(result);
      expect(back).toBeInstanceOf(DOMException);
      expect(back.message).toBe(example.message);
      expect(back.name).toBe(example.name);
    });
  });
  describe('toJSON', () => {
    it('supports DOMException', () => {
      const example = new DOMException(EXAMPLE_MESSAGE, EXAMPLE_NAME);
      const result = toJSON(example);
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = fromJSON<DOMException>(result);
      expect(back).toBeInstanceOf(DOMException);
      expect(back.message).toBe(example.message);
      expect(back.name).toBe(example.name);
    });
  });
  describe('toJSONAsync', () => {
    it('supports DOMException', async () => {
      const example = new DOMException(EXAMPLE_MESSAGE, EXAMPLE_NAME);
      const result = await toJSONAsync(Promise.resolve(example));
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = await fromJSON<Promise<DOMException>>(result);
      expect(back).toBeInstanceOf(DOMException);
      expect(back.message).toBe(example.message);
      expect(back.name).toBe(example.name);
    });
  });
  describe('crossSerialize', () => {
    it('supports DOMException', () => {
      const example = new DOMException(EXAMPLE_MESSAGE, EXAMPLE_NAME);
      const result = crossSerialize(example);
      expect(result).toMatchSnapshot();
    });
    describe('scoped', () => {
      it('supports DOMException', () => {
        const example = new DOMException(EXAMPLE_MESSAGE, EXAMPLE_NAME);
        const result = crossSerialize(example, { scopeId: 'example' });
        expect(result).toMatchSnapshot();
      });
    });
  });
  describe('crossSerializeAsync', () => {
    it('supports DOMException', async () => {
      const example = new DOMException(EXAMPLE_MESSAGE, EXAMPLE_NAME);
      const result = await crossSerializeAsync(Promise.resolve(example));
      expect(result).toMatchSnapshot();
    });
    describe('scoped', () => {
      it('supports DOMException', async () => {
        const example = new DOMException(EXAMPLE_MESSAGE, EXAMPLE_NAME);
        const result = await crossSerializeAsync(Promise.resolve(example), { scopeId: 'example' });
        expect(result).toMatchSnapshot();
      });
    });
  });
  describe('crossSerializeStream', () => {
    it('supports DOMException', async () => new Promise<void>((done) => {
      const example = new DOMException(EXAMPLE_MESSAGE, EXAMPLE_NAME);
      crossSerializeStream(Promise.resolve(example), {
        onSerialize(data) {
          expect(data).toMatchSnapshot();
        },
        onDone() {
          done();
        },
      });
    }));
    describe('scoped', () => {
      it('supports DOMException', async () => new Promise<void>((done) => {
        const example = new DOMException(EXAMPLE_MESSAGE, EXAMPLE_NAME);
        crossSerializeStream(Promise.resolve(example), {
          scopeId: 'example',
          onSerialize(data) {
            expect(data).toMatchSnapshot();
          },
          onDone() {
            done();
          },
        });
      }));
    });
  });
});
