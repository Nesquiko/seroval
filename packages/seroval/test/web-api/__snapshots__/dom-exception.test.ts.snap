// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DOMException > crossSerialize > scoped > supports DOMException 1`] = `"($R=>$R[0]=new DOMException(\\"This is an example message.\\",\\"Example\\"))($R[\\"example\\"])"`;

exports[`DOMException > crossSerialize > supports DOMException 1`] = `"$R[0]=new DOMException(\\"This is an example message.\\",\\"Example\\")"`;

exports[`DOMException > crossSerializeAsync > scoped > supports DOMException 1`] = `"($R=>$R[0]=Promise.resolve($R[1]=new DOMException(\\"This is an example message.\\",\\"Example\\")))($R[\\"example\\"])"`;

exports[`DOMException > crossSerializeAsync > supports DOMException 1`] = `"$R[0]=Promise.resolve($R[1]=new DOMException(\\"This is an example message.\\",\\"Example\\"))"`;

exports[`DOMException > crossSerializeStream > scoped > supports DOMException 1`] = `"($R=>$R[0]=_$.P())($R[\\"example\\"])"`;

exports[`DOMException > crossSerializeStream > scoped > supports DOMException 2`] = `"($R=>_$.Ps($R[0],$R[1]=new DOMException(\\"This is an example message.\\",\\"Example\\")))($R[\\"example\\"])"`;

exports[`DOMException > crossSerializeStream > supports DOMException 1`] = `"$R[0]=_$.P()"`;

exports[`DOMException > crossSerializeStream > supports DOMException 2`] = `"_$.Ps($R[0],$R[1]=new DOMException(\\"This is an example message.\\",\\"Example\\"))"`;

exports[`DOMException > serialize > supports DOMException 1`] = `"new DOMException(\\"This is an example message.\\",\\"Example\\")"`;

exports[`DOMException > serializeAsync > supports DOMException 1`] = `"Promise.resolve(new DOMException(\\"This is an example message.\\",\\"Example\\"))"`;

exports[`DOMException > toJSON > supports DOMException 1`] = `"{\\"t\\":{\\"t\\":39,\\"i\\":0,\\"s\\":\\"This is an example message.\\",\\"c\\":\\"Example\\"},\\"f\\":16383,\\"m\\":[],\\"s\\":{\\"0\\":0,\\"1\\":0}}"`;

exports[`DOMException > toJSONAsync > supports DOMException 1`] = `"{\\"t\\":{\\"t\\":12,\\"i\\":0,\\"s\\":1,\\"f\\":{\\"t\\":39,\\"i\\":1,\\"s\\":\\"This is an example message.\\",\\"c\\":\\"Example\\"}},\\"f\\":16383,\\"m\\":[],\\"s\\":{\\"0\\":0,\\"1\\":0}}"`;
